### Assignment. 산학지형 기상 정보 대비 Burned Area에 대한 EDA 및 Regression를 활용한 예측
- 데이터 집합 소스
  - 설명: https://archive.ics.uci.edu/ml/datasets/Forest+Fires
    - Features
      - 1: X - x-axis spatial coordinate within the Montesinho park map: 1 to 9 
      - 2: Y - y-axis spatial coordinate within the Montesinho park map: 2 to 9 
      - 3: month - month of the year: 'jan' to 'dec' 
      - 4: day - day of the week: 'mon' to 'sun' 
      - 5: FFMC - FFMC index from the FWI system: 18.7 to 96.20 
      - 6: DMC - DMC index from the FWI system: 1.1 to 291.3 
      - 7: DC - DC index from the FWI system: 7.9 to 860.6 
      - 8: ISI - ISI index from the FWI system: 0.0 to 56.10 
      - 9: temp - temperature in Celsius degrees: 2.2 to 33.30 
      - 10: RH - relative humidity in %: 15.0 to 100 
      - 11: wind - wind speed in km/h: 0.40 to 9.40 
      - 12: rain - outside rain in mm/m2 : 0.0 to 6.4 
    - Target 
      - area - the burned area of the forest (in ha): 0.00 to 1090.84
  - 관련 논문 및 자료
    - P. Cortez and A. Morais. A Data Mining Approach to Predict Forest Fires using Meteorological Data. In J. Neves, M. F. Santos and J. Machado Eds., New Trends in Artificial Intelligence, Proceedings of the 13th EPIA 2007 - Portuguese Conference on Artificial Intelligence, December, Guimarães, Portugal, pp. 512-523, 2007. APPIA, ISBN-13 978-989-95618-0-9. http://www3.dsi.uminho.pt/pcortez/fires.pdf
    - 논문 발췌 중요 그림
    ![attributes](./figures/fire.png)
    ![map](./figures/fire_map.png)
    - Fire Weather Index
      - https://www.frames.gov/files/6014/1576/1411/FWI-history.pdf
  - 데이터: https://archive.ics.uci.edu/ml/machine-learning-databases/forest-fires/forestfires.csv

In [1]:
import urllib2
from scipy import stats
from pandas import Series, DataFrame
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
%matplotlib inline

path = 'https://archive.ics.uci.edu/ml/machine-learning-databases/forest-fires/forestfires.csv'
raw_csv = urllib2.urlopen(path)
feature_names = ('X', 'Y', 'month', 'day', "FFMC", "DMC", 'DC', 'ISI', 'temp', 'RH', 'wind', 'rain')
target_name = 'area'
all_names = feature_names + (target_name,)
df = pd.read_csv(raw_csv)

In [2]:
df

X  Y month  day  FFMC    DMC     DC   ISI  temp  RH  wind  rain   area
0    7  5   mar  fri  86.2   26.2   94.3   5.1   8.2  51   6.7   0.0   0.00
1    7  4   oct  tue  90.6   35.4  669.1   6.7  18.0  33   0.9   0.0   0.00
2    7  4   oct  sat  90.6   43.7  686.9   6.7  14.6  33   1.3   0.0   0.00
3    8  6   mar  fri  91.7   33.3   77.5   9.0   8.3  97   4.0   0.2   0.00
4    8  6   mar  sun  89.3   51.3  102.2   9.6  11.4  99   1.8   0.0   0.00
5    8  6   aug  sun  92.3   85.3  488.0  14.7  22.2  29   5.4   0.0   0.00
6    8  6   aug  mon  92.3   88.9  495.6   8.5  24.1  27   3.1   0.0   0.00
7    8  6   aug  mon  91.5  145.4  608.2  10.7   8.0  86   2.2   0.0   0.00
8    8  6   sep  tue  91.0  129.5  692.6   7.0  13.1  63   5.4   0.0   0.00
9    7  5   sep  sat  92.5   88.0  698.6   7.1  22.8  40   4.0   0.0   0.00
10   7  5   sep  sat  92.5   88.0  698.6   7.1  17.8  51   7.2   0.0   0.00
11   7  5   sep  sat  92.8   73.2  713.0  22.6  19.3  38   4.0   0.0   0.00
12   6  5   aug  fri  63.5   70.8  665.3   0.8  17.0  72   6.7   0.0   0.00
13   6  5   sep  mon  90.9  126.5  686.5   7.0  21.3  42   2.2   0.0   0.00
14   6  5   sep  wed  92.9  133.3  699.6   9.2  26.4  21   4.5   0.0   0.00
15   6  5   sep  fri  93.3  141.2  713.9  13.9  22.9  44   5.4   0.0   0.00
16   5  5   mar  sat  91.7   35.8   80.8   7.8  15.1  27   5.4   0.0   0.00
17   8  5   oct  mon  84.9   32.8  664.2   3.0  16.7  47   4.9   0.0   0.00
18   6  4   mar  wed  89.2   27.9   70.8   6.3  15.9  35   4.0   0.0   0.00
19   6  4   apr  sat  86.3   27.4   97.1   5.1   9.3  44   4.5   0.0   0.00
20   6  4   sep  tue  91.0  129.5  692.6   7.0  18.3  40   2.7   0.0   0.00
21   5  4   sep  mon  91.8   78.5  724.3   9.2  19.1  38   2.7   0.0   0.00
22   7  4   jun  sun  94.3   96.3  200.0  56.1  21.0  44   4.5   0.0   0.00
23   7  4   aug  sat  90.2  110.9  537.4   6.2  19.5  43   5.8   0.0   0.00
24   7  4   aug  sat  93.5  139.4  594.2  20.3  23.7  32   5.8   0.0   0.00
25   7  4   aug  sun  91.4  142.4  601.4  10.6  16.3  60   5.4   0.0   0.00
26   7  4   sep  fri  92.4  117.9  668.0  12.2  19.0  34   5.8   0.0   0.00
27   7  4   sep  mon  90.9  126.5  686.5   7.0  19.4  48   1.3   0.0   0.00
28   6  3   sep  sat  93.4  145.4  721.4   8.1  30.2  24   2.7   0.0   0.00
29   6  3   sep  sun  93.5  149.3  728.6   8.1  22.8  39   3.6   0.0   0.00
..  .. ..   ...  ...   ...    ...    ...   ...   ...  ..   ...   ...    ...
487  5  4   aug  tue  95.1  141.3  605.8  17.7  26.4  34   3.6   0.0  16.40
488  4  4   aug  tue  95.1  141.3  605.8  17.7  19.4  71   7.6   0.0  46.70
489  4  4   aug  wed  95.1  141.3  605.8  17.7  20.6  58   1.3   0.0   0.00
490  4  4   aug  wed  95.1  141.3  605.8  17.7  28.7  33   4.0   0.0   0.00
491  4  4   aug  thu  95.8  152.0  624.1  13.8  32.4  21   4.5   0.0   0.00
492  1  3   aug  fri  95.9  158.0  633.6  11.3  32.4  27   2.2   0.0   0.00
493  1  3   aug  fri  95.9  158.0  633.6  11.3  27.5  29   4.5   0.0  43.32
494  6  6   aug  sat  96.0  164.0  643.0  14.0  30.8  30   4.9   0.0   8.59
495  6  6   aug  mon  96.2  175.5  661.8  16.8  23.9  42   2.2   0.0   0.00
496  4  5   aug  mon  96.2  175.5  661.8  16.8  32.6  26   3.1   0.0   2.77
497  3  4   aug  tue  96.1  181.1  671.2  14.3  32.3  27   2.2   0.0  14.68
498  6  5   aug  tue  96.1  181.1  671.2  14.3  33.3  26   2.7   0.0  40.54
499  7  5   aug  tue  96.1  181.1  671.2  14.3  27.3  63   4.9   6.4  10.82
500  8  6   aug  tue  96.1  181.1  671.2  14.3  21.6  65   4.9   0.8   0.00
501  7  5   aug  tue  96.1  181.1  671.2  14.3  21.6  65   4.9   0.8   0.00
502  4  4   aug  tue  96.1  181.1  671.2  14.3  20.7  69   4.9   0.4   0.00
503  2  4   aug  wed  94.5  139.4  689.1  20.0  29.2  30   4.9   0.0   1.95
504  4  3   aug  wed  94.5  139.4  689.1  20.0  28.9  29   4.9   0.0  49.59
505  1  2   aug  thu  91.0  163.2  744.4  10.1  26.7  35   1.8   0.0   5.80
506  1  2   aug  fri  91.0  166.9  752.6   7.1  18.5  73   8.5   0.0   0.00
507  2  4   aug  fri  91.0  166.9  752.6   7.1  25.9 

### [Mission 1] 불이 가장 많이 발생한 지역 좌표 (X, Y) 상위 5개를 제시하시오.
- numpy 및 pandas에서 제공되는 gropyby, sum, stack, sort 등의 메소드 활용 필요 

In [3]:
sdf = df[['X', 'Y', 'area']]
sdf

X  Y   area
0    7  5   0.00
1    7  4   0.00
2    7  4   0.00
3    8  6   0.00
4    8  6   0.00
5    8  6   0.00
6    8  6   0.00
7    8  6   0.00
8    8  6   0.00
9    7  5   0.00
10   7  5   0.00
11   7  5   0.00
12   6  5   0.00
13   6  5   0.00
14   6  5   0.00
15   6  5   0.00
16   5  5   0.00
17   8  5   0.00
18   6  4   0.00
19   6  4   0.00
20   6  4   0.00
21   5  4   0.00
22   7  4   0.00
23   7  4   0.00
24   7  4   0.00
25   7  4   0.00
26   7  4   0.00
27   7  4   0.00
28   6  3   0.00
29   6  3   0.00
..  .. ..    ...
487  5  4  16.40
488  4  4  46.70
489  4  4   0.00
490  4  4   0.00
491  4  4   0.00
492  1  3   0.00
493  1  3  43.32
494  6  6   8.59
495  6  6   0.00
496  4  5   2.77
497  3  4  14.68
498  6  5  40.54
499  7  5  10.82
500  8  6   0.00
501  7  5   0.00
502  4  4   0.00
503  2  4   1.95
504  4  3  49.59
505  1  2   5.80
506  1  2   0.00
507  2  4   0.00
508  1  2   0.00
509  5  4   2.17
510  6  5   0.43
511  8  6   0.00
512  4  3   6.44
513  2  4  54.29
514  7  4  11.16
515  1  4   0.00
516  6  3   0.00

[517 rows x 3 columns]

### [Mission 2] 불이 가장 많이 발생한 월 (month) 상위 2개 및 가장 많이 발생한 날 (day) 상위 2개를 제시하시오.
- numpy 및 pandas에서 제공되는 gropyby, sum, stack, sort 등의 메소드 활용 필요

In [8]:
sdf = df[['month', 'area']]
sdf

month   area
0     mar   0.00
1     oct   0.00
2     oct   0.00
3     mar   0.00
4     mar   0.00
5     aug   0.00
6     aug   0.00
7     aug   0.00
8     sep   0.00
9     sep   0.00
10    sep   0.00
11    sep   0.00
12    aug   0.00
13    sep   0.00
14    sep   0.00
15    sep   0.00
16    mar   0.00
17    oct   0.00
18    mar   0.00
19    apr   0.00
20    sep   0.00
21    sep   0.00
22    jun   0.00
23    aug   0.00
24    aug   0.00
25    aug   0.00
26    sep   0.00
27    sep   0.00
28    sep   0.00
29    sep   0.00
..    ...    ...
487   aug  16.40
488   aug  46.70
489   aug   0.00
490   aug   0.00
491   aug   0.00
492   aug   0.00
493   aug  43.32
494   aug   8.59
495   aug   0.00
496   aug   2.77
497   aug  14.68
498   aug  40.54
499   aug  10.82
500   aug   0.00
501   aug   0.00
502   aug   0.00
503   aug   1.95
504   aug  49.59
505   aug   5.80
506   aug   0.00
507   aug   0.00
508   aug   0.00
509   aug   2.17
510   aug   0.43
511   aug   0.00
512   aug   6.44
513   aug  54.29
514   aug  11.16
515   aug   0.00
516   nov   0.00

[517 rows x 2 columns]

### [Mission 3] 4대 주 요인 (temp, RH, wind, rain) 속성별 기본 통계치 및 Box Plot 산출
- numpy 및 pandas에서 제공되는 describe() 및 boxplot() 사용
- describe()가 제공하는 통계치 및 boxplot을 보면서 나름대로의 해석을 반드시 3가지이상 제시하시오.
- area와의 관계는 고려하지 말고 4가지 속성만 분석하시오.

In [12]:
sdf = df[['temp', 'RH', 'wind', 'rain']]
sdf

temp  RH  wind  rain
0     8.2  51   6.7   0.0
1    18.0  33   0.9   0.0
2    14.6  33   1.3   0.0
3     8.3  97   4.0   0.2
4    11.4  99   1.8   0.0
5    22.2  29   5.4   0.0
6    24.1  27   3.1   0.0
7     8.0  86   2.2   0.0
8    13.1  63   5.4   0.0
9    22.8  40   4.0   0.0
10   17.8  51   7.2   0.0
11   19.3  38   4.0   0.0
12   17.0  72   6.7   0.0
13   21.3  42   2.2   0.0
14   26.4  21   4.5   0.0
15   22.9  44   5.4   0.0
16   15.1  27   5.4   0.0
17   16.7  47   4.9   0.0
18   15.9  35   4.0   0.0
19    9.3  44   4.5   0.0
20   18.3  40   2.7   0.0
21   19.1  38   2.7   0.0
22   21.0  44   4.5   0.0
23   19.5  43   5.8   0.0
24   23.7  32   5.8   0.0
25   16.3  60   5.4   0.0
26   19.0  34   5.8   0.0
27   19.4  48   1.3   0.0
28   30.2  24   2.7   0.0
29   22.8  39   3.6   0.0
..    ...  ..   ...   ...
487  26.4  34   3.6   0.0
488  19.4  71   7.6   0.0
489  20.6  58   1.3   0.0
490  28.7  33   4.0   0.0
491  32.4  21   4.5   0.0
492  32.4  27   2.2   0.0
493  27.5  29   4.5   0.0
494  30.8  30   4.9   0.0
495  23.9  42   2.2   0.0
496  32.6  26   3.1   0.0
497  32.3  27   2.2   0.0
498  33.3  26   2.7   0.0
499  27.3  63   4.9   6.4
500  21.6  65   4.9   0.8
501  21.6  65   4.9   0.8
502  20.7  69   4.9   0.4
503  29.2  30   4.9   0.0
504  28.9  29   4.9   0.0
505  26.7  35   1.8   0.0
506  18.5  73   8.5   0.0
507  25.9  41   3.6   0.0
508  25.9  41   3.6   0.0
509  21.1  71   7.6   1.4
510  18.2  62   5.4   0.0
511  27.8  35   2.7   0.0
512  27.8  32   2.7   0.0
513  21.9  71   5.8   0.0
514  21.2  70   6.7   0.0
515  25.6  42   4.0   0.0
516  11.8  31   4.5   0.0

[517 rows x 4 columns]

### [Mission 4] 4대요인 및 area를 포함하여 dataframe을 얻어오고 area를 두 가지 부류로 나누어 각 4가지 속성을 비교 분석하기 
- area의 값의 편차가 너무 심하기 때문에 그러한 편차를 줄이기 위하여 numpy.log1p (자연로그) 사용
  - Calculates log(1 + x)
  - http://docs.scipy.org/doc/numpy/reference/generated/numpy.log1p.html#numpy.log1p
- 두 가지 부류로 나누는 기준은 log(1 + area) 값이 50% percentile 이상인 것과 50% percentile 이하인 것으로 정함
  - 각각을 sdf_1과 sdf_2라고 명명
- sdf_1 및 sdf_2에 대하여 4대 주 요인 (temp, RH, wind, rain) 속성별 기본 통계치 및 Box Plot 산출
  - describe()가 제공하는 통계치 및 boxplot을 보면서 나름대로의 해석을 반드시 3가지이상 제시하시오.

In [15]:
sdf = df[['temp', 'RH', 'wind', 'rain', 'area']]
sdf

temp  RH  wind  rain   area
0     8.2  51   6.7   0.0   0.00
1    18.0  33   0.9   0.0   0.00
2    14.6  33   1.3   0.0   0.00
3     8.3  97   4.0   0.2   0.00
4    11.4  99   1.8   0.0   0.00
5    22.2  29   5.4   0.0   0.00
6    24.1  27   3.1   0.0   0.00
7     8.0  86   2.2   0.0   0.00
8    13.1  63   5.4   0.0   0.00
9    22.8  40   4.0   0.0   0.00
10   17.8  51   7.2   0.0   0.00
11   19.3  38   4.0   0.0   0.00
12   17.0  72   6.7   0.0   0.00
13   21.3  42   2.2   0.0   0.00
14   26.4  21   4.5   0.0   0.00
15   22.9  44   5.4   0.0   0.00
16   15.1  27   5.4   0.0   0.00
17   16.7  47   4.9   0.0   0.00
18   15.9  35   4.0   0.0   0.00
19    9.3  44   4.5   0.0   0.00
20   18.3  40   2.7   0.0   0.00
21   19.1  38   2.7   0.0   0.00
22   21.0  44   4.5   0.0   0.00
23   19.5  43   5.8   0.0   0.00
24   23.7  32   5.8   0.0   0.00
25   16.3  60   5.4   0.0   0.00
26   19.0  34   5.8   0.0   0.00
27   19.4  48   1.3   0.0   0.00
28   30.2  24   2.7   0.0   0.00
29   22.8  39   3.6   0.0   0.00
..    ...  ..   ...   ...    ...
487  26.4  34   3.6   0.0  16.40
488  19.4  71   7.6   0.0  46.70
489  20.6  58   1.3   0.0   0.00
490  28.7  33   4.0   0.0   0.00
491  32.4  21   4.5   0.0   0.00
492  32.4  27   2.2   0.0   0.00
493  27.5  29   4.5   0.0  43.32
494  30.8  30   4.9   0.0   8.59
495  23.9  42   2.2   0.0   0.00
496  32.6  26   3.1   0.0   2.77
497  32.3  27   2.2   0.0  14.68
498  33.3  26   2.7   0.0  40.54
499  27.3  63   4.9   6.4  10.82
500  21.6  65   4.9   0.8   0.00
501  21.6  65   4.9   0.8   0.00
502  20.7  69   4.9   0.4   0.00
503  29.2  30   4.9   0.0   1.95
504  28.9  29   4.9   0.0  49.59
505  26.7  35   1.8   0.0   5.80
506  18.5  73   8.5   0.0   0.00
507  25.9  41   3.6   0.0   0.00
508  25.9  41   3.6   0.0   0.00
509  21.1  71   7.6   1.4   2.17
510  18.2  62   5.4   0.0   0.43
511  27.8  35   2.7   0.0   0.00
512  27.8  32   2.7   0.0   6.44
513  21.9  71   5.8   0.0  54.29
514  21.2  70   6.7   0.0  11.16
515  25.6  42   4.0   0.0   0.00
516  11.8  31   4.5   0.0   0.00

[517 rows x 5 columns]

### [Mission 5] 단일변수 선형 회귀분석을 통하여 가장 영향이 높은 속성 선발
- 이전 문제에서 log(1 + area) 값이 50% percentile 이상인 것과 50% percentile 으로 구분지은 것은 무시하고 전체 area 값에 대한 회귀 분석 수행
- 이전 문제에서 area의 값에 자연로그를 취하여 변환하는 것은 수행하지 마시오.
  - 원래 area 데이터에 대한 선형회귀 분석 수행
- 다음 요인들 중 area와 가장 연관성이 높은 주요 요인을 선정
  - FFMC
  - DMC
  - DC
  - ISI
  - temp
  - RH
  - wind
  - rain

In [23]:
sdf = df[['FFMC', 'DMC', 'DC', 'ISI', 'temp', 'RH', 'wind', 'rain', 'area']]
sdf

FFMC    DMC     DC   ISI  temp  RH  wind  rain   area
0    86.2   26.2   94.3   5.1   8.2  51   6.7   0.0   0.00
1    90.6   35.4  669.1   6.7  18.0  33   0.9   0.0   0.00
2    90.6   43.7  686.9   6.7  14.6  33   1.3   0.0   0.00
3    91.7   33.3   77.5   9.0   8.3  97   4.0   0.2   0.00
4    89.3   51.3  102.2   9.6  11.4  99   1.8   0.0   0.00
5    92.3   85.3  488.0  14.7  22.2  29   5.4   0.0   0.00
6    92.3   88.9  495.6   8.5  24.1  27   3.1   0.0   0.00
7    91.5  145.4  608.2  10.7   8.0  86   2.2   0.0   0.00
8    91.0  129.5  692.6   7.0  13.1  63   5.4   0.0   0.00
9    92.5   88.0  698.6   7.1  22.8  40   4.0   0.0   0.00
10   92.5   88.0  698.6   7.1  17.8  51   7.2   0.0   0.00
11   92.8   73.2  713.0  22.6  19.3  38   4.0   0.0   0.00
12   63.5   70.8  665.3   0.8  17.0  72   6.7   0.0   0.00
13   90.9  126.5  686.5   7.0  21.3  42   2.2   0.0   0.00
14   92.9  133.3  699.6   9.2  26.4  21   4.5   0.0   0.00
15   93.3  141.2  713.9  13.9  22.9  44   5.4   0.0   0.00
16   91.7   35.8   80.8   7.8  15.1  27   5.4   0.0   0.00
17   84.9   32.8  664.2   3.0  16.7  47   4.9   0.0   0.00
18   89.2   27.9   70.8   6.3  15.9  35   4.0   0.0   0.00
19   86.3   27.4   97.1   5.1   9.3  44   4.5   0.0   0.00
20   91.0  129.5  692.6   7.0  18.3  40   2.7   0.0   0.00
21   91.8   78.5  724.3   9.2  19.1  38   2.7   0.0   0.00
22   94.3   96.3  200.0  56.1  21.0  44   4.5   0.0   0.00
23   90.2  110.9  537.4   6.2  19.5  43   5.8   0.0   0.00
24   93.5  139.4  594.2  20.3  23.7  32   5.8   0.0   0.00
25   91.4  142.4  601.4  10.6  16.3  60   5.4   0.0   0.00
26   92.4  117.9  668.0  12.2  19.0  34   5.8   0.0   0.00
27   90.9  126.5  686.5   7.0  19.4  48   1.3   0.0   0.00
28   93.4  145.4  721.4   8.1  30.2  24   2.7   0.0   0.00
29   93.5  149.3  728.6   8.1  22.8  39   3.6   0.0   0.00
..    ...    ...    ...   ...   ...  ..   ...   ...    ...
487  95.1  141.3  605.8  17.7  26.4  34   3.6   0.0  16.40
488  95.1  141.3  605.8  17.7  19.4  71   7.6   0.0  46.70
489  95.1  141.3  605.8  17.7  20.6  58   1.3   0.0   0.00
490  95.1  141.3  605.8  17.7  28.7  33   4.0   0.0   0.00
491  95.8  152.0  624.1  13.8  32.4  21   4.5   0.0   0.00
492  95.9  158.0  633.6  11.3  32.4  27   2.2   0.0   0.00
493  95.9  158.0  633.6  11.3  27.5  29   4.5   0.0  43.32
494  96.0  164.0  643.0  14.0  30.8  30   4.9   0.0   8.59
495  96.2  175.5  661.8  16.8  23.9  42   2.2   0.0   0.00
496  96.2  175.5  661.8  16.8  32.6  26   3.1   0.0   2.77
497  96.1  181.1  671.2  14.3  32.3  27   2.2   0.0  14.68
498  96.1  181.1  671.2  14.3  33.3  26   2.7   0.0  40.54
499  96.1  181.1  671.2  14.3  27.3  63   4.9   6.4  10.82
500  96.1  181.1  671.2  14.3  21.6  65   4.9   0.8   0.00
501  96.1  181.1  671.2  14.3  21.6  65   4.9   0.8   0.00
502  96.1  181.1  671.2  14.3  20.7  69   4.9   0.4   0.00
503  94.5  139.4  689.1  20.0  29.2  30   4.9   0.0   1.95
504  94.5  139.4  689.1  20.0  28.9  29   4.9   0.0  49.59
505  91.0  163.2  744.4  10.1  26.7  35   1.8   0.0   5.80
506  91.0  166.9  752.6   7.1  18.5  73   8.5   0.0   0.00
507  91.0  166.9  752.6   7.1  25.9  41   3.6   0.0   0.00
508  91.0  166.9  752.6   7.1  25.9  41   3.6   0.0   0.00
509  91.0  166.9  752.6   7.1  21.1  71   7.6   1.4   2.17
510  91.0  166.9  752.6   7.1  18.2  62   5.4   0.0   0.43
511  81.6   56.7  665.6   1.9  27.8  35   2.7   0.0   0.00
512  81.6   56.7  665.6   1.9  27.8  32   2.7   0.0   6.44
513  81.6   56.7  665.6   1.9  21.9  71   5.8   0.0  54.29
514  81.6   56.7  665.6   1.9  21.2  70   6.7   0.0  11.16
515  94.4  146.0  614.7  11.3  25.6  42   4.0   0.0   0.00
516  79.5    3.0  106.7   1.1  11.8  31   4.5   0.0   0.00

[517 rows x 9 columns]

### [Mission 6] 가장 영향이 높은 요인 두 개로 다변수 선형 회귀분석 수행
- 전체 517개의 데이터에 대한 area 값과 선형회귀 분석 결과인 predicted area 값을 모두 한꺼번에 출력하여 비교하시오.
- 선형 회귀분석 성능이 좋은지 자신의 의견을 제시하시오.

### [Mission 7] 가장 영향이 높은 요인 두 개로 다변수 로지스틱 회귀분석 수행
- area의 값의 편차가 너무 심하기 때문에 그러한 편차를 줄이기 위하여 numpy.log1p (자연로그) 사용
  - Calculates log(1 + x)
  - http://docs.scipy.org/doc/numpy/reference/generated/numpy.log1p.html#numpy.log1p
- 새로운 Categorical Variable로서 0 및 1을 지니는 'fire' 컬럼 생성
- 두 가지 부류로 나누는 기준은 log(1 + area) 값이 50% percentile 이상인 것과 50% percentile 이하인 것으로 정함
  - log(1 + area) 값이 50% percentile 이상이면 'fire' 컬럼 값이 1
  - log(1 + area) 값이 50% percentile 이하이면 'fire' 컬럼 값이 0
- 로지스틱 회귀분석에 의한 분류의 정확도를 최종적으로 출력하시오.  